### DISCUSSION TITLE
How to read a value generated inside a long running task?

### DISCUSSION THREAD

**Discussion by MindSwipe** [bevyengine/bevy#2506] [Q&A] [answered]
Using `AsyncComputeTaskPool` I can spawn a CPU intensive `Task` which may span across multiple frames, but I can't read the result of the task back. Common usage for this is something like decoupling procedural world generation from the framerate. Building on that this is what I currently have:

```rust
// Just placeholder for this question
struct WorldGenTask(Option<Task<WorldGenResult>>);
struct WorldGenResult(u32);

fn gen_world_system(
  pool: Res<AsyncComputeTaskPool>,
  mut task: ResMut<WorldGenTask>,
  input: Res<Input<KeyCode>>,
) {
  if task.0.is_none() && input.just_pressed(KeyCode::Space) {
    let t = pool.spawn(temp());
    *task = WorldGenTask(Some(t));
  }
}

fn gen_world() -> WorldGenResult {
  // Do some CPU intensive work generating the world
  WorldGenResult(42)
}
```

This spawns the task on the task pool and runs it, but how can I now get the value generated inside a system to spawn my components?

**Comment by TheRawMeatball** [comment, marked as answer]
Look here: https://github.com/bevyengine/bevy/blob/main/examples/async_tasks/async_compute.rs

This example was added after 0.5, but it should still get the point across :)

**Reply by TheNeikos** [reply]
Man, I’ve been using queues for this. But this is much smarter!
